src/easydmp/lib/api/pagination.py

```diff
@@ -0,0 +1,31 @@
+from rest_framework import pagination
+
+
+__all__ = [
+    'PageNumberPagination',
+    'ToggleablePageNumberPagination',
+]
+
+
+class PaginationToggleMixin:
+    "Paginate if `page_size` is an integer in GET query paramaters"
+
+    def paginate_queryset(self, queryset, request, view=None):
+        if not hasattr(self, 'page_size_query_param'):
```

Comment:
Alternatively not have the mixin to avoid strange ifs and buts like this? The mixin makes some assumption about a field in the subtype, I actually struggled to understand this code, which should be trivial.

Reply:
Yeah, I guess I abstracted too soon. YAGNI I guess.

Reply:
I've removed the mixin and simiplified a little.